src/inmanta/data/model.py

```diff
@@ -287,11 +285,47 @@ def level_from_enum_attribute(cls, v: object) -> object:
             try:
                 return const.LogLevel[v]
             except KeyError:
-                raise ValueError(f"Invalid enum value {v}. Valid values: { ','.join([x.name for x in const.LogLevel]) }")
+                raise ValueError(f"Invalid enum value {v}. Valid values: {','.join([x.name for x in const.LogLevel])}")
         return v
 
     level: const.LogLevel
     msg: str
     args: List[Optional[ArgumentTypes]] = []
     kwargs: Dict[str, Optional[ArgumentTypes]] = {}
     timestamp: datetime.datetime
+
+
+class ResourceIdDetails(BaseModel):
+    resource_type: ResourceType
+    agent: str
+    attribute: str
+    resource_id_value: str
+
+
+class LatestReleasedResource(BaseModel):
+    resource_id: ResourceIdStr
+    resource_version_id: ResourceVersionIdStr
+    id_details: ResourceIdDetails
+    requires: List[ResourceVersionIdStr]
+    status: const.ResourceState
```

Comment:
isn't there a new status here for orphans? (i.e. shouldn't this be another type)

Reply:
I assumed it would be added to the `ResourceState` enum, and handled similarly to other statuses. We can also add a property here to indicate this if it's more important.
Or do you mean that it should be determined by this query, if a resource is orphaned or not?

Reply:
Let's not touch the ResourceState enum, that is a bit too central to the whole orchestrator.